### DISCUSSION TITLE
memory caching behaves unexpectedly

### DISCUSSION THREAD

**Discussion by ftalbrecht** [pymor/pymor#1132] [Q&A] [answered]
I known I ran across this previously but do not really remember what we did or decided. I have a `fom` with a `numpy`-based solution space of about 10.000 DoFs and

```python
fom.enable_caching('memory')
```

that I solve for about a hundred different parameters using the usual

```python
mus = parameter_space.sample_uniformly(100) # imagine a 1d parameter space
U = fom.solution_space.empty()
for mu in mus:
    U.append(fom.solve(mu))
assert len(U) == len(mus)
```
which works as expected.
However, running the same code again, the assertion fails and I have `len(mus)=100` and `len(U)=61`.
When investigating, I find that for some of these `mu`s, the result of `fom.solve(mu)` is simply an empty vector array:
```python
NumpyVectorArray([], NumpyVectorSpace(10201, id='STATE'))
```

Disabling caching or changing the backend to `disk` resolves the issue. While it seems that the storage of the `memory` cache backend is simply depleted, I feel that we should handle the situation better. In particular, *the result of `fom.solve` should not depend on the caching*!

Am I missing something, @sdrave?

**Comment by sdrave** [comment, marked as answer]
This is clearly a bug. See #1133.
